### PR TITLE
feat: end-block process orderbook plugin

### PR DESCRIPTION
**Pull request by p0mvn**

### CODE REVIEW

File: app/sidecar_query_server.go [hunk 1]
```diff
@@ -212,6 +213,14 @@ func NewSideCarQueryServer(appCodec codec.Codec, config domain.Config, logger lo
 			return nil, err
 		}
 
+		// Register ingest block end process plugins
+		// TODO: move to config.
```
Comment (p0mvn) [inline app/sidecar_query_server.go]: Note: config TODO

File: ingest/usecase/plugins/orderbookfiller/ordebook_filler_ingest_plugin.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		routerUseCase: routerUseCase,
+		tokensUseCase: tokensUseCase,
+	}
+}
+
+// ProcessEndBlock implements domain.EndBlockProcessPlugin.
+func (o *orderbookFillerIngestPlugin) ProcessEndBlock(ctx context.Context, blockHeight uint64, metadata domain.BlockPoolMetadata) error {
+	o.logger.Info("processing end block in orderbook filler ingest plugin", zap.Uint64("block_height", blockHeight))
+	return nil
+}
```
Comment (p0mvn) [inline ingest/usecase/plugins/orderbookfiller/ordebook_filler_ingest_plugin.go]: Note: write core logic here. This is called async at the end of every block.

File: ingest/usecase/ingest_usecase.go [hunk 3]
```diff
@@ -43,6 +43,9 @@ type ingestUseCase struct {
 	// Worker that computes candidate routes for all tokens.
 	candidateRouteSearchWorker domain.CandidateRouteSearchDataWorker
 
+	// endBlockProcessPlugins are the plugins to execute at the end of the block.
+	endBlockProcessPlugins []domain.EndBlockProcessPlugin
```
Comment (AlpinYukseloglu) [inline ingest/usecase/ingest_usecase.go]: I love that you made this a generic primitive!

File: .vscode/launch.json [hunk 4]
```diff
@@ -10,6 +10,11 @@
       "request": "launch",
       "mode": "auto",
       "program": "${workspaceFolder}/app",
+      "env": {
+        "OSMOSIS_KEYRING_PATH":"/root/.osmosisd/keyring-test",
+        "OSMOSIS_KEYRING_PASSWORD": "test",
+	    "OSMOSIS_KEYRING_KEY_NAME": "local.info"
+      },
```
Comment (p0mvn) [inline .vscode/launch.json]: @niccoloraspa you need to have a key with name `local` created under test keyring and have these env vars set

File: ingest/usecase/plugins/orderbookfiller/ordebook_filler_ingest_plugin.go [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+				o.logger.Debug("failed to process orderbook", zap.Error(err))
+			}
+		case <-time.After(10 * time.Second):
+			o.logger.Error("timed out processing orderbook")
+			return err
+		}
+	}
+
+	o.logger.Info("processed end block in orderbook filler ingest plugin", zap.Uint64("block_height", blockHeight))
+	return nil
```
Comment (p0mvn) [inline ingest/usecase/plugins/orderbookfiller/ordebook_filler_ingest_plugin.go]: @niccoloraspa we read the canonical orderbooks and then try to detect cyclic arb opportunities against top SQS route for each

File: ingest/usecase/plugins/orderbookfiller/ordebook_filler_ingest_plugin.go [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+	if err != nil {
+		return err
+	}
+
+	errTolerance := osmomath.ErrTolerance{
+		MultiplicativeTolerance: sdk.NewDecWithPrec(5, 2),
+	}
+
+	//
+	difference := errTolerance.Compare(baseInCoin.Amount, cyclicArbQuote.GetAmountOut())
```
Comment (p0mvn) [inline ingest/usecase/plugins/orderbookfiller/ordebook_filler_ingest_plugin.go]: @niccoloraspa arb detection logic

File: ingest/usecase/plugins/orderbookfiller/osmosis_swap.go [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+func (o *orderbookFillerIngestPlugin) swapExactAmountIn(tokenIn sdk.Coin, route []domain.RoutablePool) (response *coretypes.ResultBroadcastTx, txbody string, err error) {
```
Comment (p0mvn) [inline ingest/usecase/plugins/orderbookfiller/osmosis_swap.go]: @niccoloraspa swap message signing logic

**Pull request closed without merging**